func_decor.py

```diff
@@ -0,0 +1,64 @@
+from time import time
+
+registry = {}
+
+
+def wrapper(func):
+    def timer(*args, **kwargs):
+        start = time()
+        res = func(*args, **kwargs)
+        stop = time()
+        registry[func.__name__] = (
+            "result: {result}; function execution time: {timer}".format
+            (result=res, timer=stop - start))
+        return res
+
+    return timer
+
+
+@wrapper
+def spam(*args):
+    return {arg: arg ** 2 for arg in args}  # spam = wrapper(spam)
+
+
+@wrapper
+def get_ranges(lst):
```

Comment:
your function doesn't work correctly and has wrong output

Reply:
solved.